addons/account/tests/test_early_payment_discount.py

```diff
@@ -61,6 +61,25 @@ def test_early_payment_end_date(self):
                     fields.Date.from_string('2019-01-11') or False
                 )
 
+    def test_invoice_report_without_invoice_date(self):
+        """
+        Ensure that an invoice with an early discount payment term
+        and no invoice date can be previewed or printed.
+        """
+        out_invoice = self.env['account.move'].create([{
+            'move_type': 'out_invoice',
+            'invoice_payment_term_id': self.early_pay_10_percents_10_days.id,
+            'invoice_line_ids': [Command.create({
+                'name': 'line1',
+            })]
+        }])
+
+        # Assert that the invoice date is not set
+        self.assertEqual(out_invoice.invoice_date, False)
```

Comment:
I won't ask you to change it but for your information, you can use `self.assertFalse()` :wink: